src/Symfony/Component/Workflow/TransitionBlockerList.php

```diff
@@ -0,0 +1,196 @@
+            $string .= $blocker."\n";
+        }
+
+        return $string;
+    }
+
+    /**
+     * Adds a transition blocker to this list.
+     *
+     * @param TransitionBlocker $blocker
+     */
+    public function add(TransitionBlocker $blocker): void
+    {
+        $this->blockers[] = $blocker;
+    }
+
+    /**
+     * Merges an existing blocker list into this list.
+     *
+     * @param TransitionBlockerList $otherList
+     */
+    public function addAll(self $otherList): void
+    {
+        foreach ($otherList as $blocker) {
+            $this->blockers[] = $blocker;
+        }
+    }
+
+    /**
+     * Returns the blocker at a given offset.
+     *
+     * @param int $offset The offset of the blocker
+     *
+     * @return TransitionBlocker The blocker
+     *
+     * @throws \OutOfBoundsException if the offset does not exist
+     */
+    public function get($offset): TransitionBlocker
+    {
+        if (!isset($this->blockers[$offset])) {
+            throw new \OutOfBoundsException(sprintf('The offset "%s" does not exist.', $offset));
+        }
+
+        return $this->blockers[$offset];
+    }
+
+    /**
+     * Returns whether the given offset exists.
+     *
+     * @param int $offset The blocker offset
+     *
+     * @return bool Whether the offset exists
+     */
+    public function has($offset): bool
+    {
+        return isset($this->blockers[$offset]);
+    }
+
+    /**
+     * Sets a blocker at a given offset.
+     *
+     * @param int               $offset  The blocker offset
+     * @param TransitionBlocker $blocker The blocker
+     */
+    public function set($offset, TransitionBlocker $blocker): void
+    {
+        $this->blockers[$offset] = $blocker;
+    }
+
+    /**
+     * Removes a blocker at a given offset.
+     *
+     * @param int $offset The offset to remove
+     */
+    public function remove($offset): void
+    {
+        unset($this->blockers[$offset]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return \ArrayIterator|TransitionBlocker[]
+     */
+    public function getIterator()
+    {
+        return new \ArrayIterator($this->blockers);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        return count($this->blockers);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetExists($offset)
+    {
+        return $this->has($offset);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetGet($offset)
+    {
+        return $this->get($offset);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetSet($offset, $blocker)
+    {
+        if (null === $offset) {
+            $this->add($blocker);
+        } else {
+            $this->set($offset, $blocker);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetUnset($offset)
+    {
+        $this->remove($offset);
+    }
+
+    /**
+     * Creates iterator for blockers with specific code.
```

Comment:
I agree, that the doc block is wrong. It's a copy&paste failure from `ConstraintViolationList.php`. I've removed the doc block, because it's obvious what that method does.

It's useful to keep this method as non-internal, because then devs can run conditional logic (i.e. `if` statements) in their code, basing on presence of a specific TransitionBlocker in the TransitionBlockerList, that's carried over from the `Workflow::apply()` method. I, for example, am going to check for the presence of `TransitionBlocker::REASON_TRANSITION_NOT_APPLICABLE` in my code and handle this case slightly differently, than other blockers.

It's also useful to keep this method as a getter and not as a hasser, because the getter form allows devs to see, whether the specific blocker is there and also retrieve this blocker, while the hasser only allows to see, whether the blocker is there. It's useful to be able to retrieve the blocker, because it may contain `$parameters`, that matter to the dev.

This method returns the first blocker, because I assumed, that no one would try to block a transition with the same reason twice. Although possible, I don't know why someone would need to do that. I certainly don't need that.

In summary: can I keep the method as it is, please (except for the wrong doc block of course)?